moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -0,0 +1,250 @@
+                    if (!state.isNullOrEmpty()) {
+                        MoEAnalyticsHelper.setUserAttribute(
+                            application.applicationContext,
+                            USER_TRAIT_USER_ADDRESS_STATE,
+                            state,
+                            instanceId
+                        )
+                    }
+                }
+            }
+            val location = payload.traits[USER_TRAIT_USER_LOCATION]
+            if (location != null && !location.jsonObject.isEmpty()) {
+                MoEAnalyticsHelper.setUserAttribute(
+                    application.applicationContext, USER_ATTRIBUTE_USER_LOCATION, GeoLocation(
+                        location.jsonObject.getDouble(USER_TRAIT_USER_LOCATION_LATITUDE) ?: 0.0,
+                        location.jsonObject.getDouble(USER_TRAIT_USER_LOCATION_LONGITUDE) ?: 0.0
+                    ), instanceId
+                )
+            }
+            Logger.print { "$tag identify(): out" }
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag identify(): " }
+        }
+
+        return payload
+    }
+
+    override fun reset() {
+        try {
+            super.reset()
+            Logger.print { "$tag reset(): in" }
+            logoutUser(application.applicationContext, instanceId)
+            Logger.print { "$tag reset(): out" }
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag reset(): " }
+        }
+    }
+
+    override fun track(payload: TrackEvent): BaseEvent {
+        try {
+            super.track(payload)
+            Logger.print { "$tag track(): in" }
+            if (payload.properties.isNotEmpty()) {
+                integrationHelper.trackEvent(
+                    payload.event, payload.properties.toJSONObject(), instanceId
+                )
+            } else {
+                integrationHelper.trackEvent(payload.event, JSONObject(), instanceId)
+            }
+            Logger.print { "$tag track(): out" }
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag track(): " }
+        }
+        return payload
+    }
+
+    override fun version(): String {
+        return BuildConfig.MOENGAGE_SEGMENT_KOTLIN_VERSION
+    }
+
+    private fun trackAnonymousId() {
+        try {
+            Logger.print { "$tag trackAnonymousId() : in" }
+            val anonymousId = analytics.storage.read(Storage.Constants.AnonymousId)
+            Logger.print(LogLevel.DEBUG) { "$tag trackAnonymousId() : $anonymousId" }
+            integrationHelper.trackAnonymousId(
+                anonymousId,
+                instanceId
+            )
+            Logger.print { "$tag trackAnonymousId() : out" }
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag trackAnonymousId(): " }
+        }
+    }
+
+    private fun filterOutNonNullTraits(traits: Traits): Map<String, Any> {
```

Comment:
`removeTraitsWithNullValues()` would be a better name

Reply:
Done